lib/puppet/provider/package/dnfmodule.rb

```diff
@@ -34,14 +34,21 @@ def self.prefetch(packages)
 
   def self.instances
     packages = []
-    cmd = "#{command(:dnf)} module list --installed -d 0 -e #{error_level}"
+    cmd = "#{command(:dnf)} module list --enabled -d 0 -e #{error_level}"
     execute(cmd).each_line do |line|
-      next unless line =~ /\[i\][, ]/  # get rid of non-package lines (including last Hint line)
-      line.gsub!(/\[[de]\]/, '')  # we don't care about default/enabled flags
+      next unless line =~ /\[[ei]\][, ]/  # get rid of non-package lines (including last Hint line)
```

Comment:
the comment is misleading, as the regexp selects lines containing install/enabled modules

Reply:
well... it skips lines that are not packages, since dnf prints some extraneous information, like the first and last two lines here:

```
localmirror-appstream
Name                                  Stream                        Profiles                                         Summary                                                               
389-ds                                1.4 [e]                                                                        389 Directory Server (base)                                           
httpd                                 2.4 [d][e]                    common [d] [i], devel, minimal                   Apache HTTP Server                                                    
python27                              2.7 [d][e]                    common [d]                                       Python programming language, version 2.7                              
python36                              3.6 [d][e]                    common [d], build                                Python programming language, version 3.6                              
ruby                                  2.5 [d][e]                    common [d]                                       An interpreter of object-oriented scripting language                  
satellite-5-client                    1.0 [d][e]                    common [d], gui                                  Red Hat Satellite 5 client packages                                   

Hint: [d]efault, [e]nabled, [x]disabled, [i]nstalled
```

Reply:
My understanding is that it skips all lines not containing `[i]` or `[e]` followed by a ` ` or `,` , it will skip even lines with modules that are not installed or enabled. This behavior is good, but I still find the comment misleading
```
389-ds               1.4                                    389 Directory Server (base)                                     
ant                  1.10 [d]   common [d]                  Java build tool                                                 
container-tools      1.0        common [d]                  Common tools and dependencies for container runtimes            
container-tools      rhel8 [d]  common [d]                  Common tools and dependencies for container runtimes            
```

Reply:
> it will skip even lines with modules that are not installed or enabled

since we're running `dnf module list` with the `--enabled` parameter we should only get enabled modules here, but I will reword the comment